app/components/work_package_relations_tab/index_component.rb

```diff
@@ -74,24 +86,31 @@ def render_header(border_box, title, items)
 
   def render_items(border_box, items)
     items.each do |item|
-      related_work_package_id = find_related_work_package_id(item)
-      data_attribute = nil
-      if related_work_package_id.to_s == @scroll_to_id
-        data_attribute = {
-          controller: "work-packages--relations-tab--scroll",
-          application_target: "dynamic",
-          "work-packages--relations-tab--scroll-target": "scrollToRow"
-        }
-      end
       border_box.with_row(
         test_selector: row_test_selector(item),
-        data: data_attribute
+        data: data_attribute(item)
       ) do
         yield(item)
       end
     end
   end
 
+  def data_attribute(item)
+    if scroll_to?(item)
+      {
+        controller: "work-packages--relations-tab--scroll",
+        application_target: "dynamic",
+        "work-packages--relations-tab--scroll-target": "scrollToRow"
+      }
+    end
+  end
+
+  def scroll_to?(item)
+    relation_to_scroll_to \
+      && item.id == relation_to_scroll_to.id \
+      && item.instance_of?(relation_to_scroll_to.class)
+  end
```

Comment:
As `relation_to_scroll_to` could be either a `Relation` or a `WorkPackage` (child relation), the class had to be checked too to prevent scrolling to a wrong element because of a similar id (unlikely but possible).

I added a commit directly to fix it (and extract methods).